src/java.base/share/classes/java/net/URLDecoder.java

```diff
@@ -144,24 +144,28 @@ public static String decode(String s, String enc) throws UnsupportedEncodingExce
      * Decodes an {@code application/x-www-form-urlencoded} string using
      * a specific {@linkplain Charset Charset}.
      * The supplied charset is used to determine
-     * what characters are represented by any consecutive sequences of the
-     * form "<i>{@code %xy}</i>".
+     * what characters are represented by any consecutive escape sequences of
+     * the form "<i>{@code %xy}</i>".
      * <p>
      * <em><strong>Note:</strong> The <a href=
      * "http://www.w3.org/TR/html40/appendix/notes.html#non-ascii-chars">
      * World Wide Web Consortium Recommendation</a> states that
      * UTF-8 should be used. Not doing so may introduce
      * incompatibilities.</em>
+     * <p>
+     * If any consecutive well-formed escape sequences cannot
+     * be decoded as a sequence of characters in the supplied {@code Charset}
+     * {@linkplain java.nio.charset.CharsetDecoder##cae the replacement character} will be used.
      *
      * @implNote This implementation will throw an {@link java.lang.IllegalArgumentException}
-     * when illegal strings are encountered.
+     * when malformed escape sequences are encountered.
      *
      * @param s the {@code String} to decode
      * @param charset the given charset
      * @return the newly decoded {@code String}
      * @throws NullPointerException if {@code s} or {@code charset} is {@code null}
-     * @throws IllegalArgumentException if the implementation encounters illegal
-     * characters
+     * @throws IllegalArgumentException if the implementation encounters malformed
+     * escape sequences
```

Comment:
The method specifies that it throws IAE, the implNote seems to be saying the same thing, do I read this correctly? I'm wondering if the implNote can be removed.

Reply:
> That's a good point. I see that there's another `decode(String, String)` method above in this file that has the same old `@implNote` but not `@throws`. Maybe the implNote should be removed there too and the `@throws` added. Not sure it's worth touching the first `@Deprecated decode(String)` method though. Opinions?

Since we are editing this method descriptions then it's probably best to add the throws IAE to the other 2-arg decode method. I suppose the 1-arg/deprecated decode method should document the exception too, doesn't need to be done in this PR of course.

Reply:
I'd be happy to change all in this PR if there are no objections

Reply:
Yes - let's fix it all here.